README.md

```diff
@@ -5,7 +5,7 @@
 A pairing-based threshold cryptosystem for collaborative decryption and
 signatures.
 
-The `threshold_crypto` crate provides constructors for encrypted message handling. It utilizes the [`pairing`](https://crates.io/crates/pairing) elliptic curve library to create and enable reconstruction of public and private key shares.
+The `threshold_crypto` crate provides constructors for encrypted message handling. It utilizes the [`pairing`](https://crates.io/crates/pairing) elliptic curve library to create and enable reconstruction of public and private key shares. The elliptic curve arithmetic and pairings are based on [Boneh-Lynn-Shacham](https://www.iacr.org/archive/asiacrypt2001/22480516.pdf) (BLS12) signatures.
```

Comment:
If I understand correctly, both the [original BLS signatures](https://www.iacr.org/archive/asiacrypt2001/22480516.pdf) and [Alexandra Boldyreva's _threshold_ crypto](https://eprint.iacr.org/2002/118.pdf) variant of them are schemes that can be implemented on top of either [a pairing function](https://en.wikipedia.org/wiki/Pairing-based_cryptography) or a [Gap Diffie-Hellman  (GDH) group](https://en.wikipedia.org/wiki/XDH_assumption), which often (but not always) are just two different ways to look at the same mathematical object. Both of _those_ usually are based on an elliptic curve.
The BLS paper seems to be formulated in the pairing setting, and the Boldyreva paper in GDH setting. Our implementation is pairing-based but includes _threshold_ crypto.

So I think we should cite both papers here. Maybe something like this?

```markdown
The `threshold_crypto` crate provides cryptographic keys with methods for signing and encrypting messages, as well as key sets for _threshold_ signatures and threshold encryption.
The threshold signature scheme is described in [the paper by Alexandra Boldyreva](Alexandra Boldyreva), which extends [Boneh-Lynn-Shacham](https://www.iacr.org/archive/asiacrypt2001/22480516.pdf) signatures to the threshold setting. The encryption part uses the [scheme by Baek and Zhang](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.119.1717&rep=rep1&type=pdf).
Our implementation is based on the [`pairing`](https://crates.io/crates/pairing) elliptic curve library.
```

Note that the `pairing` library coincidentally uses a [Barreto-Lynn-Scott curve](https://z.cash/blog/new-snark-curve/), which _also_ happens to be abbreviated as "BLS". :grin: Not sure whether we should mention that?